### PR TITLE
CA-322749: Make toolstack work with old Nvidia host drivers

### PR DESCRIPTION
**Pull request by liulinC**
This commit make xenopsd work with old Nvidia drivers as followings,
When VM start, xenopsd will check whether the config_file element exists
in the vm metadata, if yes, then xenopsd think it starts the VM in the legnecy
mode and prepare the config_file arguments for demu, otherwise, the
type_id for the vGPU is used. Xapi is responsible to check whether the
installed host driver support multiple vGPU and set the metadata.

Signed-off-by: Lin Liu <lin.liu@citrix.com>

### CODE REVIEW

File: xc/device.ml [hunk 1]
```diff
@@ -2017,14 +2017,13 @@ module Dm_Common = struct
                |> String.concat ",")
           in
           match x.implementation with
-          | Nvidia {virtual_pci_address; type_id = Some type_id; uuid = Some uuid; extra_args} ->
-            make [type_id; Xcp_pci.string_of_address virtual_pci_address; uuid; extra_args]
+          (* 1. Upgrade case, migrate from a old host with old vGPU having config_path
+           * 2. Legency case, run with old Nvidia host driver *)
           | Nvidia {virtual_pci_address; config_file = Some config_file; extra_args} ->
```
Comment (lindig) [inline xc/device.ml]: I think both cases should have a `debug` log line:
* `debug "NVidia vGPU config: using config file %s" confile_file`
* `debug "NVidia vGPU config: using type id %s" type_id`
If possible, including the UUID of the vGPU would be good.

Comment (liulinC) [reply xc/device.ml]: updated.

**Comment by robhoes** [comment]
Please raise another PR to remove the now-unused function `nvidia_compat_lookup`.

**Comment by liulinC** [comment]
@robhoes , yes I methoned that in the PR of xapi part, will remove it after this fix.

**Pull request merged**